File: JS/move-zeroes-to-end.js
function moveZeroes(arr) {
    let insertPos = 0;
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] !== 0) {
            arr[insertPos++] = arr[i];
        }
    }
    while (insertPos < arr.length) arr[insertPos++] = 0;
    return arr;
}

const lines = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n');
const n = parseInt(lines[0]);
const arr = lines[1].trim().split(/\s+/).map(Number).slice(0, n);
console.log(moveZeroes(arr).join(' '));

File: C++/move-zeroes-to-end.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n; cin >> n;
    vector<long long> arr(n);
    for (auto &x : arr) cin >> x;

    int insertPos = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i] != 0) arr[insertPos++] = arr[i];
    }
    while (insertPos < n) arr[insertPos++] = 0;

    for (int i = 0; i < n; i++) cout << arr[i] << " \n"[i == n - 1];
    return 0;
}